test/cluster/mzcompose.py

```diff
@@ -2546,3 +2548,83 @@ def worker(c: Composition, i: int) -> None:
         assert (
             p99 < p99_limit
         ), f"p99 is {p99:.2f}s, should be less than {p99_limit:.2f}s"
+
+
+def workflow_test_profile_fetch(c: Composition) -> None:
+    """
+    Test fetching memory and CPU profiles via the internal HTTP
+    endpoint.
+    """
+
+    c.down(destroy_volumes=True)
+    c.up("materialized")
+    c.up("clusterd1")
+
+    def curl(container: str, *args: str) -> int:
```

Comment:
I would use Python's requests library instead, will probably make errors easier to understand.

Reply:
Had to figure out how to get the right ports, but this is indeed nicer!